sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py

```diff
@@ -481,6 +481,61 @@ def send_message( # type: ignore
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace
+    def receive_one_message(self, **kwargs):
+        # type: (Optional[Any]) -> QueueMessage
+        """Removes one message from the front of the queue.
+
+        When the message is retrieved from the queue, the response includes the message
+        content and a pop_receipt value, which is required to delete the message.
+        The message is not automatically deleted from the queue, but after it has
+        been retrieved, it is not visible to other clients for the time interval
+        specified by the visibility_timeout parameter.
+
+        If the key-encryption-key or resolver field is set on the local service object, the message will be
+        decrypted before being returned.
+
+        :keyword int visibility_timeout:
+            If not specified, the default value is 0. Specifies the
+            new visibility timeout value, in seconds, relative to server time.
+            The value must be larger than or equal to 0, and cannot be
+            larger than 7 days. The visibility timeout of a message cannot be
+            set to a value later than the expiry time. visibility_timeout
+            should be set to a value smaller than the time-to-live value.
+        :keyword int timeout:
+            The server timeout, expressed in seconds.
+        :return:
+            Returns a message from the Queue.
+        :rtype: ~azure.storage.queue.QueueMessage
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/queue_samples_message.py
+                :start-after: [START receive_one_message]
+                :end-before: [END receive_one_message]
+                :language: python
+                :dedent: 12
+                :caption: Receive one message from the queue.
+        """
+        visibility_timeout = kwargs.pop('visibility_timeout', None)
+        timeout = kwargs.pop('timeout', None)
+        self._config.message_decode_policy.configure(
+            require_encryption=self.require_encryption,
+            key_encryption_key=self.key_encryption_key,
+            resolver=self.key_resolver_function)
+        try:
+            message = self._client.messages.dequeue(
+                number_of_messages=1,
+                visibilitytimeout=visibility_timeout,
+                timeout=timeout,
+                cls=self._config.message_decode_policy,
+                **kwargs
+            )
+            wrapped_message = QueueMessage._from_generated(message[0]) # pylint: disable=protected-access
```

Comment:
the `dequeue` can return empty list. What would happen in that case?

Reply:
I completely forgot to consider that edge case, good catch!